module_utils/network_lsr/ethtool.py

```diff
@@ -0,0 +1,54 @@
+# SPDX-License-Identifier: BSD-3-Clause
+
+import array
+import binascii
+import struct
+import fcntl
+import socket
+
+
+ETHTOOL_GPERMADDR = 0x00000020
+SIOCETHTOOL = 0x8946
+
+
+def get_perm_addr(ifname):
+    """
+    Return the Permanent address value for the specified interface using the
+    ETHTOOL_GPERMADDR ioctl command.
+
+    Please for further documentation, see:
+    https://github.com/torvalds/linux/blob/master/include/uapi/linux/ethtool.h#L734
+    https://github.com/torvalds/linux/blob/master/include/uapi/linux/ethtool.h#L1388
+    https://git.kernel.org/pub/scm/network/ethtool/ethtool.git/tree/ethtool.c#n4172
+    """
+    try:
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        sockfd = sock.fileno()
+        ifname = ifname.encode("utf-8")
+        ecmd = array.array("B", struct.pack("II6s", ETHTOOL_GPERMADDR, 6, b"\x00" * 6))
+        ifreq = struct.pack("16sP", ifname, ecmd.buffer_info()[0])
+
+        fcntl.ioctl(sockfd, SIOCETHTOOL, ifreq)
+        try:
+            res = ecmd.tobytes()
+        except AttributeError:
+            # tobytes() is not available in python2
+            res = ecmd.tostring()
+        _, _, perm_addr = struct.unpack("II6s", res)
+        perm_addr = _decode_perm_addr(perm_addr)
+    except IOError:
+        perm_addr = None
+    finally:
+        sock.close()
+
+    return perm_addr
+
+
+def _decode_perm_addr(perm_addr_bytes):
+    if hasattr(perm_addr_bytes, "hex"):
+        perm_addr = perm_addr_bytes.hex()
+    else:
+        perm_addr = binascii.hexlify(perm_addr_bytes)
+    perm_addr = ":".join(f + s for f, s in zip(perm_addr[::2], perm_addr[1::2]))
```

Comment:
Please consider renaming `f` and `s` to  `first_nibble` and `second_nibble`

Reply:
Done, thanks!